tests/js/web/TableDetailComponentTest.js

```diff
@@ -0,0 +1,118 @@
+import $ from 'jquery';
+import TableDetailComponent from '../../../js/TableDetailComponent'
+
+describe('TableDetailComponent', () => {
+	const clickEvent = $.Event('click');
+	let $html;
+	let $body;
+	let $tableHTML;
+	let tableDetailComponent;
+	let tableDetailComponentWithoutHTML;
+
+	beforeEach(() => {
+		$html = $('<html></html>');
```

Comment:
We need to be vigilant when using this pattern, now the browser environment test suite is fixed things like this will break it. If we replace the contents of the `<html/>` element it will prevent the tests from running. Does this _need_ to be a `<html/>` node, or could it live as a `<div/>` for the tests?

Reply:
alternatively maybe we could cache the HTML contents in the before each and restore it in the after each, I tend to favour the first option though.